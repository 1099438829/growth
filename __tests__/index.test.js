
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Growth from '../src/index';

jest.mock('react-native-fs', () => ({
  writeFile: jest.fn(() => Promise.resolve()),
  ExternalStorageDirectoryPath: 'package-path',
}));


describe('test index', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Growth />,
    );

    const treeJson = tree.toJSON();
    expect(treeJson).toMatchSnapshot();
  });
});
