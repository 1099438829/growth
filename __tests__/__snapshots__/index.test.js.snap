// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test index renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      },
      Object {
        "backgroundColor": "#000",
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      pointerEvents="auto"
      style={
        Object {
          "backgroundColor": "white",
          "bottom": 0,
          "left": 0,
          "opacity": 1,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateX": 0,
            },
            Object {
              "translateY": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              pointerEvents="auto"
              removeClippedSubviews={false}
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "opacity": 1,
                  },
                ]
              }
            >
              <View
                onLayout={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "flex": 1,
                    },
                    Object {
                      "backgroundColor": "#000",
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <View
                    onMoveShouldSetResponder={[Function]}
                    onMoveShouldSetResponderCapture={[Function]}
                    onResponderEnd={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderReject={[Function]}
                    onResponderRelease={[Function]}
                    onResponderStart={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    onStartShouldSetResponderCapture={[Function]}
                    pointerEvents="auto"
                    style={
                      Object {
                        "backgroundColor": "white",
                        "bottom": 0,
                        "left": 0,
                        "opacity": 1,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                        "transform": Array [
                          Object {
                            "translateX": 0,
                          },
                          Object {
                            "translateY": 0,
                          },
                        ],
                      }
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "flex": 1,
                          },
                          Object {
                            "backgroundColor": "#E9EBEE",
                            "paddingBottom": 51,
                            "paddingTop": 64,
                          },
                        ]
                      }
                    />
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "backgroundColor": "#03a9f4",
                      "borderBottomColor": "#828287",
                      "borderBottomWidth": 0,
                      "height": 64,
                      "left": 0,
                      "paddingTop": 0,
                      "position": "absolute",
                      "right": 0,
                      "shadowColor": "rgba(0,0,0, .2)",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": 0,
                      },
                      "shadowOpacity": 1,
                      "shadowRadius": 1,
                      "top": 0,
                    }
                  }
                >
                  <View>
                    <View
                      style={
                        Object {
                          "left": 0,
                          "marginTop": 10,
                          "position": "absolute",
                          "right": 0,
                          "top": 20,
                        }
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        lineBreakMode="tail"
                        numberOfLines={1}
                        style={
                          Object {
                            "alignSelf": "center",
                            "color": "#ffffff",
                            "fontFamily": "HelveticaNeue",
                            "fontSize": 14,
                            "fontWeight": "bold",
                            "left": 0,
                            "opacity": 1,
                            "right": 0,
                            "textAlign": "center",
                            "width": 180,
                          }
                        }
                      >
                        Growth
                      </Text>
                    </View>
                    <View
                      style={
                        Array [
                          Object {
                            "height": 37,
                            "left": 2,
                            "padding": 8,
                            "position": "absolute",
                            "top": 20,
                          },
                          Object {
                            "alignItems": "flex-start",
                          },
                        ]
                      }
                    >
                      <View
                        accessibilityComponentType={undefined}
                        accessibilityLabel={undefined}
                        accessibilityTraits={undefined}
                        accessible={true}
                        hasTVPreferredFocus={undefined}
                        hitSlop={undefined}
                        isTVSelectable={true}
                        onLayout={undefined}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Array [
                            Object {
                              "backgroundColor": "transparent",
                            },
                            Array [
                              false,
                              false,
                              false,
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "transparent",
                                "justifyContent": "center",
                              },
                              undefined,
                            ],
                          ]
                        }
                        testID={undefined}
                        tvParallaxProperties={undefined}
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={false}
                          ellipsizeMode="tail"
                          style={
                            Array [
                              Object {
                                "color": "#fff",
                                "fontSize": 24,
                              },
                              Array [
                                Object {
                                  "backgroundColor": "transparent",
                                },
                                undefined,
                              ],
                              Object {
                                "fontFamily": "Ionicons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                            ]
                          }
                        >
                          
                        </Text>
                      </View>
                    </View>
                    <View
                      accessibilityComponentType={undefined}
                      accessibilityLabel={undefined}
                      accessibilityTraits={undefined}
                      accessible={true}
                      hitSlop={undefined}
                      isTVSelectable={true}
                      onLayout={undefined}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "height": 37,
                          "opacity": 1,
                          "padding": 8,
                          "position": "absolute",
                          "right": 2,
                          "top": 22,
                        }
                      }
                      testID="rightNavButton"
                      tvParallaxProperties={undefined}
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {
                              "color": "rgb(0, 122, 255)",
                              "fontSize": 17,
                              "textAlign": "right",
                            },
                            Object {
                              "color": "#ffffff",
                              "fontFamily": "HelveticaNeue",
                              "fontSize": 14,
                              "fontWeight": "bold",
                            },
                            Object {
                              "color": "#ffffff",
                              "fontFamily": "HelveticaNeue",
                              "fontSize": 14,
                              "fontWeight": "bold",
                            },
                          ]
                        }
                      >
                        购买纸质版
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "bottom": 0,
                  "flexDirection": "row",
                  "height": 50,
                  "justifyContent": "center",
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                },
                Object {
                  "backgroundColor": "#000",
                },
              ]
            }
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              isTVSelectable={true}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ffffff",
                  "borderTopColor": "#D0D1D5",
                  "borderTopWidth": 1,
                  "flex": 1,
                  "height": 50,
                  "justifyContent": "center",
                  "opacity": 1,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View>
                <View
                  onPress={undefined}
                  style={
                    Array [
                      false,
                      false,
                      false,
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      },
                      undefined,
                    ]
                  }
                  underlayColor="white"
                >
                  <Text
                    accessible={true}
                    allowFontScaling={false}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "#03a9f4",
                          "fontSize": 24,
                        },
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                          },
                          undefined,
                        ],
                        Object {
                          "fontFamily": "Ionicons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                      ]
                    }
                  >
                    
                  </Text>
                </View>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#03a9f4",
                    }
                  }
                >
                  Growth
                </Text>
              </View>
            </View>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              isTVSelectable={true}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ffffff",
                  "borderTopColor": "#D0D1D5",
                  "borderTopWidth": 1,
                  "flex": 1,
                  "height": 50,
                  "justifyContent": "center",
                  "opacity": 1,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View>
                <View
                  onPress={undefined}
                  style={
                    Array [
                      false,
                      false,
                      false,
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      },
                      undefined,
                    ]
                  }
                  underlayColor="white"
                >
                  <Text
                    accessible={true}
                    allowFontScaling={false}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "#767676",
                          "fontSize": 24,
                        },
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                          },
                          undefined,
                        ],
                        Object {
                          "fontFamily": "Ionicons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                      ]
                    }
                  >
                    
                  </Text>
                </View>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#767676",
                    }
                  }
                >
                  探索
                </Text>
              </View>
            </View>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              isTVSelectable={true}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ffffff",
                  "borderTopColor": "#D0D1D5",
                  "borderTopWidth": 1,
                  "flex": 1,
                  "height": 50,
                  "justifyContent": "center",
                  "opacity": 1,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View>
                <View
                  onPress={undefined}
                  style={
                    Array [
                      false,
                      false,
                      false,
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      },
                      undefined,
                    ]
                  }
                  underlayColor="white"
                >
                  <Text
                    accessible={true}
                    allowFontScaling={false}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "#767676",
                          "fontSize": 24,
                        },
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                          },
                          undefined,
                        ],
                        Object {
                          "fontFamily": "Ionicons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                      ]
                    }
                  >
                    
                  </Text>
                </View>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#767676",
                    }
                  }
                >
                  技能树
                </Text>
              </View>
            </View>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              isTVSelectable={true}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ffffff",
                  "borderTopColor": "#D0D1D5",
                  "borderTopWidth": 1,
                  "flex": 1,
                  "height": 50,
                  "justifyContent": "center",
                  "opacity": 1,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View>
                <View
                  onPress={undefined}
                  style={
                    Array [
                      false,
                      false,
                      false,
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      },
                      undefined,
                    ]
                  }
                  underlayColor="white"
                >
                  <Text
                    accessible={true}
                    allowFontScaling={false}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "#767676",
                          "fontSize": 24,
                        },
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                          },
                          undefined,
                        ],
                        Object {
                          "fontFamily": "Ionicons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                      ]
                    }
                  >
                    
                  </Text>
                </View>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#767676",
                    }
                  }
                >
                  社区
                </Text>
              </View>
            </View>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              isTVSelectable={true}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ffffff",
                  "borderTopColor": "#D0D1D5",
                  "borderTopWidth": 1,
                  "flex": 1,
                  "height": 50,
                  "justifyContent": "center",
                  "opacity": 1,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View>
                <View
                  onPress={undefined}
                  style={
                    Array [
                      false,
                      false,
                      false,
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      },
                      undefined,
                    ]
                  }
                  underlayColor="white"
                >
                  <Text
                    accessible={true}
                    allowFontScaling={false}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "#767676",
                          "fontSize": 24,
                        },
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                          },
                          undefined,
                        ],
                        Object {
                          "fontFamily": "Ionicons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                      ]
                    }
                  >
                    
                  </Text>
                </View>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#767676",
                    }
                  }
                >
                  练习
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
